Objects/frameobject.c

```diff
@@ -676,6 +684,60 @@ framelocalsproxy_setdefault(PyObject* self, PyObject *const *args, Py_ssize_t na
     return result;
 }
 
+static PyObject*
+framelocalsproxy_pop(PyObject* self, PyObject *const *args, Py_ssize_t nargs)
+{
+    if (nargs < 1 || nargs > 2) {
+        PyErr_SetString(PyExc_TypeError, "pop expected 1 or 2 arguments");
+        return NULL;
+    }
+
+    PyObject *key = args[0];
+    PyObject *default_value = NULL;
+
+    if (nargs == 2) {
+        default_value = args[1];
+    }
+
+    PyFrameObject *frame = ((PyFrameLocalsProxyObject*)self)->frame;
+
+    int i = framelocalsproxy_getkeyindex(frame, key, false);
+    if (i == -2) {
+        return NULL;
+    }
+
+    if (i >= 0) {
+        PyErr_SetString(PyExc_KeyError, "cannot remove local variables from FrameLocalsProxy");
+        return NULL;
+    }
+
+    PyObject* result = NULL;
+
+    if (frame->f_extra_locals == NULL) {
+        if (default_value != NULL) {
+            return Py_XNewRef(default_value);
+        } else {
+            PyErr_Format(PyExc_KeyError, "'%R'", key);
```

Comment:
A tiny optimization to consider (though, it's more complex--feel free to reject this):
```suggestion
            PyObject *repr = PyObject_Repr(key);
            if (repr == NULL) {
                return NULL;
            }
            PyErr_SetObject(PyExc_KeyError, repr);
```

Reply:
I just realized I could've used `_PyErr_SetKeyError(key)` like `dictobject.c`. I'll switch to that.

Reply:
Oh, that sounds better. I wasn't aware of that, TIL!

Reply:
So I switched to `_PyErr_SetKeyError(key)` when the key does not exist. However, for local variables cases, I think `RuntimeError` would fit better. One important reason was `KeyError` actually expects the "key" to be the explanation string, and giving a reason is a bit weird. Also in this case we have the key, we just can't remove it, so `RuntimeError` might be better.